### ISSUE TITLE
A user can manually add a site to "Always open in [container]"

### ISSUE THREAD

**Issue by mlncn** [mozilla/multi-account-containers#1227]
In short: This is a convenience and becomes necessary in the case of the Temporary Containers plugin protecting against sites that don't belong in a container opening in that container, and a site that redirects to another domain and back (as for a poorly done login; see many bank web sites).

I need a way in Firefox to manually add sites to Multi-account containers.  I just upgraded from the built-in version to this extension, so apologies if i missed it— i see that at least this extension gives a list of sites that are assigned to a container.

This is something i wanted before— it's faster to type a bunch of URLs into a textarea than it is to open a container, type one URL into the address bar, then click the Multi-Account Containers icon, and checkmark "Always open in ThisContainer"  ... and repeat, and repeat.

But now i need it (for certain definitions of need).

If a site is redirected to another site as for login, and temporary containers is on to protect us from just these things, this can prevent us from ever logging into that site:

1. MyBankWithBadWebsite.com in Banking container (set to always open in).
2. When i log in, it redirects to MyBankWithBadWebsite-Login.net, which opens in a temporary container, then redirects me back to the original banking web site.
3. I'm not logged in, and have no way to add MyBankWithBadWebsite-Login.net to the Banking container because it redirects me if i go there directly, also.

But I need to add MyBankWithBadWebsite-Login.net to the Banking container so that the login can complete successfully.

- Operating System + Version: PopOS (Ubuntu)
- Firefox Version: 60.0.2 (64-bit)
- Relevant installed Add-ons + Version + Enabled/Disabled-Status:
Duplicate Tabs Closer	3.1.5	true	
Firefox Multi-Account Containers	6.0.0	true
Form History Control (II)	2.0.3.2	true	
Switch Container	0.8	true	
Temporary Containers	0.90	true

**Comment by da2x** [comment]
Same problem with different use case. I type in https://adsense.com but can’t add it to the open-in-container list because it redirects to a different domain (google.com).

**Comment by falolaf** [comment]
Would this addon be to any help:
https://addons.mozilla.org/sv-SE/firefox/addon/containerise/

**Comment by cjkent** [comment]
This would make it very useful for working with AWS. This plugin allows you to be logged into the console for multiple AWS accounts simultaneously. 

The URL for the AWS console login is different for each account, e.g.

https://account-name.signin.aws.amazon.com/console

but that URL immediately redirects to a URL that is the same for all accounts, so there's no way to check the "Always open in ..." checkbox.

You can work around it with some very fiddly hacking of the config file, but it would be so much easier if you could just manually associate a URL with a container tab.

**Comment by mastazi** [comment]
I have the same problem with youtu.be and many other URL shorteners. I am wondering if there is a workaround.

**Comment by rahulshw** [comment]
This a problem for me as well. The SSO auth site or url shortener services redirect immediately before you can move your mouse to the checkbox of "Always open in".

Till this issue is worked around @mastazi 's solution work very well but there is a bit of problem if my url shortner does not have a "." in it(i.e. instead of youtu.be if its just yout), which in my case was. If you type "yout" in the address bar and hit enter, the browser will try to reach www.yout.com by default. One can disable this smartness of the browser by following the steps [here](https://support.mozilla.org/en-US/kb/search-web-address-bar#w_domain-guessing) and then following, @mastazi 's solution.

**Comment by kendallcorner** [comment]
Linking to #473 , but I don't think it's the same issue.

**Comment by PauloGDPeixoto** [comment]
> Update to my previous comment: I did it by turning off my internet connection temporarily. This way the URL youtu.be showed an error page and from there I could add it to my container. Now clicking on Youtu.be links no longer results in duplicated tabs opening.

There's a workaround in this comment that doesn't require disconnecting your network: https://github.com/mozilla/multi-account-containers/issues/473#issuecomment-376095758

**Comment by diego898** [comment]
I believe #1500 will fix both this issue and #473

**Comment by smithbm2316** [comment]
> Update to my previous comment: I did it by turning off my internet connection temporarily. This way the URL youtu.be showed an error page and from there I could add it to my container. Now clicking on Youtu.be links no longer results in duplicated tabs opening.

@mastazi THANK YOU SO MUCH FOR THIS COMMENT. I have spent the last hour and a half trying to figure out a way to add my company's redirect email link that opens up a Gmail login page to always open in a specific container, since I have other Gmail accounts that I use and can't just add "mail.google.com" to only open in that container. You have blessed my day.

**Comment by CanRau** [comment]
@diego898 not sure how this #1500 will fix manually adding a url?
Though #1688 addresses just that 😍

**Comment by Tamarindo94** [comment]
This is a fundamental feature that would solve many problems :)

**Comment by 1JOe** [comment]
> 
> 
> > Update to my previous comment: I did it by turning off my internet connection temporarily. This way the URL youtu.be showed an error page and from there I could add it to my container. Now clicking on Youtu.be links no longer results in duplicated tabs opening.
> 

This workaround still works and helped a lot with drive.google.com. My google container couldn't handle the redirects to the temporary tab google opens.

**Comment by grahamperrin** [comment]
This seems to be one of a number of duplicates of #839

**Comment by grahamperrin** [comment]
▶ #839

@mlncn please, can you close this duplicate issue? Thank you.

**Comment by t0mmy44-h** [comment]
This feature is required to use MAC with Amazon AWS IAM sign-in links. Because these add a subdomain `<account-name>.signin.amazonaws.com` that redirects to `signin.amazonaws.com`, users can't use these links if MAC has a container configured for AWS.

**Comment by kunaltyagi** [comment]
Both #1500 and #1688 have seen no work for a long time. Any ideas when this feature would land?

**Comment by s3-m-bocian** [comment]
Can you Guys point me to the path of the plug-in's config file(s)?
I'd love to manually add numerous pages.
Many thanks in advance

**Comment by Spitfire1900** [comment]
There's a crude workaround to this that fits @t0mmy44-h 's exact usecase with AWS.

1. Create a new container
2. Open `about:config`
3. Set `network.http.redirection-limit=0`, congrats you've just disabled redirects in Firefox!
4. Navigate to `<account-name>.signin.amazonaws.com`
5. You will get a "The page isn’t redirecting properly" error, but crucialy _will still be on the `<account-name>.signin.amazonaws.com` page._
6. In the right-hand side of the URL bar click the multi-account container icon to open the "Always open this site in..." dialog
7. Select the container created earlier.
8. Reset `network.http.redirection-limit` using the "Reset" button on the right-hand side of `about:config`.

**Comment by mattmcadoo** [comment]
Setting the `network.http.redirection-limit=0` parameter in about:config doesn't seem to work for authenticated Google sites [#2532 .](https://github.com/mozilla/multi-account-containers/issues/2532)  Any chance of this getting a resolved or a better kludge workaround before this issue hits its 6th anniversary?

**Comment by Spitfire1900** [comment]
@mattmcadoo , is the window location being replaced by JS? If so try this:

1. Open a new tab and open up Dev Tools' Debugger tab.
2. Select XHR Breakpoints > Pause on any URL
3. Go to the URL you want to add, the page should be paused when JS tries to replace the page.
4. In the right-hand side of the URL bar click the multi-account container icon to open the "Always open this site in..." dialog

**Comment by mattmcadoo** [comment]
@Spitfire1900  No, it's bouncing to different subdomains to go through their sign process.  The only way I can sign in is to disable the whitelist-only (Limit to designated sites) option for the container, signin, and then re-enable it.

Is the reason why there cannot be a `*.domain.tld` or `.domain.tld` is because of the usage of SQLite for all of the config settings?

**Comment by Spitfire1900** [comment]
>disable the whitelist-only
This is a known issue with any site that uses OIDC or SAML to do logins.

**Comment by nabeelr** [comment]
This issue has been around for 8 years... Will this ever get fixed? There should be a way to manually add items and remove items to the "Manage site list" menu. You can already click to clear a site's cookies, and remove specific sites, why not add?

<img width="354" height="594" alt="Image" src="https://github.com/user-attachments/assets/cec894a6-0818-4b2d-8a6f-ebd4366da506" />

<img width="354" height="299" alt="Image" src="https://github.com/user-attachments/assets/b89860e7-c9f4-4f65-a265-4b92b2871aac" />

Such a simple option, and the work arounds are so ridiculous, especially when you have the "limit to designated sites" option. Tell me: How can I limit it to designated sites, when I can't designate the site?!

This is basic functionality that's missing because of a very common condition that was not considered as a plausible edge case.

This is particularly annoying, since apparently the time was taken to put an advertisement for Mozilla's VPN into the menu, but actually give us functionality we need? Nope.

**Comment by kunaltyagi** [comment]
TBH, this seems "simple" enough. If the maintainers accept, I could try adding this feature.

I've never developed an extension, so if someone shares a good starting guide, that'd also help (along with the moz blessings)
